### PR TITLE
Add AWS documentation (DynamoDB, Redshift)

### PR DESCRIPTION
**Pull request by tsotnet**
Signed-off-by: Tsotne Tabidze <tsotne@tecton.ai>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
Document AWS Provider (including DynamoDB & Redshift)
```

### CODE REVIEW

File: docs/reference/offline-stores/redshift.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+registry: data/registry.db
+provider: aws
+offline_store:
+  type: redshift
+  region: us-west-2
+  cluster_id: feast-cluster
+  database: feast-database
+  user: redshift-user
+  s3_staging_location: s3://feast-bucket/redshift
+  iam_role: arn:aws:iam::123456789012:role/redshift_s3_access_role
```
Comment (woop) [inline docs/reference/offline-stores/redshift.md:26]: We don't have to quote this configuration?

Comment (tsotnet) [reply docs/reference/offline-stores/redshift.md:26]: Doesn't seem to be necessary

**Review by woop — APPROVED** [review]

**Comment by woop** [comment]
Assuming TODOs will come in another PR.

**Pull request merged**